src/commands/delete.js

```diff
@@ -0,0 +1,54 @@
+const utils = require('../utils');
+
+const _delete = (context, appOrVersion, version) => {
+  const isDeletingVersion = appOrVersion === 'version';
+  if (isDeletingVersion && !version) {
+    const message = 'Error: No version - provide a version like "1.0.0"...';
+    return Promise.reject(new Error(message));
+  }
+  return utils.checkCredentials()
+    .then(() => utils.getLinkedApp())
+    .then((app) => {
+      const deletePrepMessage = isDeletingVersion
+        ? `Preparing to delete version ${version} of your app "${app.title}".\n`
+        : 'Preparing to delete all versions of your app "${app.title}".\n';
+      context.line(deletePrepMessage);
+      const url = isDeletingVersion ? `/apps/${app.id}/versions/${version}` : `/apps/${app.id}`;
```

Comment:
Sorry, I wasn't explicit. Splitting up the code here would be helpful for readability/maintainability. The moving of funcs to a Model could be done at a later time. Splitting this up would remove all the ternaries.